src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1185 @@
+    replacements.add(null);
+
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof VariableDeclarationExpr) {
+        // Create and return a list of possible replacement statements.
+        VariableDeclarationExpr vdExpr = (VariableDeclarationExpr) exp;
+
+        // Simplify right hand side to zero-equivalent value:  0, false, or null.
+        Statement rhsZeroValStmt = rhsAssignZeroValue(vdExpr);
+        if (rhsZeroValStmt != null) {
+          replacements.add(rhsZeroValStmt);
+        }
+
+        // Simplify right hand side to a value that was previously found
+        // in a passing assertion.
+        Statement rhsAssertValStmt = rhsAssignValueFromPassingAssertion(vdExpr, primitiveValues);
+        if (rhsAssertValStmt != null) {
+          replacements.add(rhsAssertValStmt);
+        }
+
+        // Simplify statement by removing the left hand side.
+        Statement lhsRemovalStmt = removeLeftHandSideSimplification(vdExpr);
+        if (lhsRemovalStmt != null) {
+          replacements.add(lhsRemovalStmt);
+        }
+      }
+    }
+    return replacements;
+  }
+
+  /**
+   * If {@code currStmt} is a statement that is an assertTrue statement using an '==' operator,
+   * store the value associated with the variable in the {@code primitiveValues} map.
+   *
+   * @param currStmt a statement
+   * @param primitiveValues a map of variable names to variable values; modified if {@code currStmt}
+   *     is a passing assertion, asserting a variable's value.
+   */
+  private static void storeValueFromAssertion(
+      Statement currStmt, Map<String, String> primitiveValues) {
+    // Check if the statement is an assertion regarding a value that can be
+    // used in a simplification later on.
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof MethodCallExpr) {
+        MethodCallExpr mCall = (MethodCallExpr) exp;
+        // Check if the method call is an assertTrue statement
+        if (mCall.getName().equals("assertTrue")) {
+          // Is an assertTrue statement.
+          List<Expression> mArgs = mCall.getArgs();
+          if (mArgs.size() == 1) {
+            // Retrieve and store the value associated with the
+            // variable in the assertion.
+            Expression mExp = mArgs.get(0);
+            List<Node> children = mExp.getChildrenNodes();
+
+            // Check that the expression only has two nodes.
+            if (children.size() != 2) {
+              return;
+            }
+            String mExpStr = mExp.toString();
+
+            // Take the substring without the first and second expression nodes.
+            String firstExp = children.get(0).toString();
+            int secondExpIndex = mExpStr.lastIndexOf(children.get(1).toString());
+            mExpStr = mExpStr.substring(firstExp.length(), secondExpIndex);
```

Comment:
String manipulations are ugly and error-prone.  Please get the operator directly from the AST rather than doing String manipulation.  It's a bit more work to understand the API well enough to use it, but worth it in terms of robustness (and efficiency).